lib/pdf/core/renderer.rb

```diff
@@ -216,10 +217,12 @@ def render_xref(output)
       # Write out the PDF Trailer, as per spec 3.4.4
       #
       def render_trailer(output)
+        trailer_id = PDF::Core::ByteString.new(SecureRandom.random_bytes(16))
```

Comment:
With 'deterministic' are you thinking of hashing some document content? Because the PDF spec (10.3 File Identifiers) suggest to use the current time beside some other data. But that would not solve the problem with the idempotent test you mentioned somewhere else.